From aabbe5544459cdc8fc830c3ecb1dfda66023473c Mon Sep 17 00:00:00 2001
From: Jukka Laitinen <jukka.laitinen@intel.com>
Date: Fri, 5 May 2017 11:26:01 +0300
Subject: [PATCH] Fix compilation for 4.8 and above kernels

Some macros have changed in kernel 4.8. This modifies the
backport kconfig to use the original kernel versions of these
macros.

Signed-off-by: Jukka Laitinen <jukka.laitinen@intel.com>
---
 backport-include/linux/kconfig.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/backport-include/linux/kconfig.h b/backport-include/linux/kconfig.h
index 4fe2a10..c6d8e39 100644
--- a/backport-include/linux/kconfig.h
+++ b/backport-include/linux/kconfig.h
@@ -5,6 +5,8 @@
 #include_next <linux/kconfig.h>
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,8,0)
+
 #ifndef __ARG_PLACEHOLDER_1
 #define __ARG_PLACEHOLDER_1 0,
 #define config_enabled(cfg) _config_enabled(cfg)
@@ -16,6 +18,7 @@
  * 3.1 - 3.3 had a broken version of this, so undef
  * (they didn't have __ARG_PLACEHOLDER_1)
  */
+
 #undef IS_ENABLED
 #define IS_ENABLED(option) \
         (config_enabled(option) || config_enabled(option##_MODULE))
@@ -24,6 +27,8 @@
 #undef IS_BUILTIN
 #define IS_BUILTIN(option) config_enabled(option)
 
+#endif // < KERNEL_VERSION(4,8,0)
+
 #ifndef IS_REACHABLE
 /*
  * IS_REACHABLE(CONFIG_FOO) evaluates to 1 if the currently compiled
-- 
2.7.4

